tools/cmake/tvos.toolchain.cmake

```diff
@@ -8,10 +8,15 @@ set_common_xcode_attributes()
 
 set(REALM_SKIP_SHARED_LIB ON)
 
+# CMake special-cases the compiler detection for iOS in a way that happens
+# to also work for tvOS, so use the iphoneos base SDK to opt-in to that and
+# then override the platform to tvOS below
+set(CMAKE_OSX_SYSROOT "iphoneos")
```

Comment:
Maybe the comment is explaining the wrong thing.

Before this change, we were telling CMake that we were building for macOS, and then adjusting the build settings in the Xcode project to make it actually build for watchOS. This worked largely by coincidence until CMake 3.9, which fixed a bug related to targeting multiple platforms and made it so that macOS targets would always actually use the object files built for macOS. Since our macOS target wasn't building files for macOS, this broke our project.

The correct fix here would be to do `set(CMAKE_OSX_SYSROOT "tvos")`, but that doesn't work because when detecting the compiler identity CMake tries to build a command-line tool for all platforms except iphoneos. The non-hacky fix for this is to completely override all of the compiler identity checking and such and explicitly set all the settings, but that's a whole bunch of code.

Setting the platform to iphoneos doesn't run into the same problems as setting it to macos did because for multi-arch platforms CMake makes fewer assumptions about what arches exist and will actually be built, since as far as it knows the armv7k we build for watchOS against a watchOS SDK could be an iOS arch, while they are apparently comfortable with hardcoding the idea that macOS is x86_64.